inc/ticket.class.php

```diff
@@ -830,4 +845,66 @@ static function filter_actors(array $params = []): array {
 
        return $params;
     }
+
+    public static function addToTimeline($options)
+    {
+        //Grous in the tickets' entity that are not assigned to the current ticket
+        $groups = (new Group())->find([
+            'is_assign' => 1,
+            'NOT' => [
+                // Group currently assigned to the ticket
+                'id' => new QuerySubQuery([
+                    'SELECT' => 'groups_id',
+                    'FROM'   => Group_Ticket::getTable(),
+                    'WHERE'  => [
+                        'tickets_id' => $options['item']->getID(),
+                        'type' => CommonITILActor::ASSIGN,
+                    ]
+                ])
+            ],
+            // Restrict to ticket entity
+            getEntitiesRestrictCriteria(
+                Group::getTable(),
+                '',
+                $options['item']->fields['entities_id']
+            )
+        ]);
+
+        $itemtypes = [];
+
+        if (!empty($groups) && $options['item'] instanceof Ticket) {
+            $itemtypes['escalation'] = [
+                'type' => 'PluginEscaladeTicket',
+                'class' => 'action-escalation',
+                'icon' => 'ti ti-arrow-up',
+                'label' => __('Escalate', 'escalade'),
+                'item' => new self()
+            ];
+        }
+
+        return $itemtypes;
+    }
+
+    public function showForm($ID, $options = [])
+    {
+        global $CFG_GLPI;
+        $groupsAssign = new Group_Ticket();
+        $groups = $groupsAssign->find([
+            'tickets_id' => $options["parent"]->getID(),
+            'type' => CommonITILActor::ASSIGN,
+        ]);
+        $used = array();
+        foreach ($groups as $grp) {
+
+            $used[$grp['groups_id']] = $grp['groups_id'];
+
+        }
+
+        $options['action'] = PLUGIN_ESCALADE_WEBDIR . '/front/ticket.form.php';
+        TemplateRenderer::getInstance()->display('@escalade/escalade_form.html.twig', [
+            'params' => $options,
+            'subitem' => $options["parent"],
+            'used' => $used
+        ]);
```

Comment:
1. You should pass `action` to the template directly.
2. Passing options to the template seems useless.

```suggestion
        TemplateRenderer::getInstance()->display('@escalade/escalade_form.html.twig', [
            'action' => PLUGIN_ESCALADE_WEBDIR . '/front/ticket.form.php',
            'subitem' => $options["parent"],
            'used' => $used
        ]);
```